diff --git a/dist/builder-manager/index.cjs b/dist/builder-manager/index.cjs
index fad8c8ab214e4b5f62283c1934ea8e06b6fb87c2..3859d07c8c335e26e4f1456a6840f631fe3f7192 100644
--- a/dist/builder-manager/index.cjs
+++ b/dist/builder-manager/index.cjs
@@ -1999,12 +1999,8 @@ var Ki = /^\/($|\?)/, ie, D, he = /* @__PURE__ */ o(async (e) => {
     tsconfig: r,
     legalComments: "external",
     plugins: [
-      (0, Ut.default)({
-        process: require.resolve("process/browser.js"),
-        util: require.resolve("util/util.js"),
-        assert: require.resolve("browser-assert")
-      }),
       oe(Ht.globalsModuleInfoMap),
+      require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')),
       (0, Jt.pnpPlugin)()
     ],
     banner: {
@@ -2013,10 +2009,13 @@ var Ki = /^\/($|\?)/, ie, D, he = /* @__PURE__ */ o(async (e) => {
     footer: {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
+    inject: [b.resolve('node-stdlib-browser/helpers/esbuild/shim')],
     define: {
+      process: "process",
       "process.env": JSON.stringify(a),
       ...(0, Bt.stringifyProcessEnvs)(a),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/builder-manager/index.js b/dist/builder-manager/index.js
index bca8212bb8ca5b948e2539aa0aa09c928522a479..959d85bdc3a6a1033b388bef4590b69c0adbdeb7 100644
--- a/dist/builder-manager/index.js
+++ b/dist/builder-manager/index.js
@@ -2011,12 +2011,8 @@ var ur = /^\/($|\?)/, G, C, _t = /* @__PURE__ */ o(async (e) => {
     tsconfig: r,
     legalComments: "external",
     plugins: [
-      (0, Ct.default)({
-        process: O.resolve("process/browser.js"),
-        util: O.resolve("util/util.js"),
-        assert: O.resolve("browser-assert")
-      }),
       Z(mr),
+      require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')),
       (0, Lt.pnpPlugin)()
     ],
     banner: {
@@ -2026,9 +2022,11 @@ var ur = /^\/($|\?)/, G, C, _t = /* @__PURE__ */ o(async (e) => {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
     define: {
+      process: "process",
       "process.env": JSON.stringify(a),
       ...pr(a),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/common/index.cjs b/dist/common/index.cjs
index 4adcc83615b4b460a1a1a9d837ea7af85fbb2f18..57a02dc8126b38a5e587a879ee03cefe45c6ac5f 100644
--- a/dist/common/index.cjs
+++ b/dist/common/index.cjs
@@ -11686,7 +11686,7 @@ function hi(r) {
     yC = !0, i({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: !1,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
diff --git a/dist/common/index.js b/dist/common/index.js
index 9cd55dbfff70967592b1e29e7b3b0257fcaf707e..2abe40a4dd47721e16c0cf844b112f03471cf371 100644
--- a/dist/common/index.js
+++ b/dist/common/index.js
@@ -11617,7 +11617,7 @@ function Mo(r) {
     Fn = !0, i({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: !1,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
