diff --git a/dist/builder-manager/index.cjs b/dist/builder-manager/index.cjs
index e714cdb06e9a1cea756d4275189beeff0239df9d..3c36a2a7b66540b859fb0a1cca7a936b2b813cd3 100644
--- a/dist/builder-manager/index.cjs
+++ b/dist/builder-manager/index.cjs
@@ -2985,12 +2985,8 @@ var ze, he, St = /* @__PURE__ */ a(async (e) => {
     tsconfig: n,
     legalComments: "external",
     plugins: [
-      (0, Fi.default)({
-        process: require.resolve("process/browser.js"),
-        util: require.resolve("util/util.js"),
-        assert: require.resolve("browser-assert")
-      }),
       Xe(Oi.globalsModuleInfoMap),
+      require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')),
       (0, Pi.pnpPlugin)()
     ],
     banner: {
@@ -2999,10 +2995,13 @@ var ze, he, St = /* @__PURE__ */ a(async (e) => {
     footer: {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
+    inject: [b.resolve('node-stdlib-browser/helpers/esbuild/shim')],
     define: {
+      process: "process",
       "process.env": JSON.stringify(i),
       ...(0, xi.stringifyProcessEnvs)(i),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/builder-manager/index.js b/dist/builder-manager/index.js
index a4f1ef5f313e83e455423a470d1fdaaad0dbed30..2689239e663ad173ecc00a6beaeef0cfec90d127 100644
--- a/dist/builder-manager/index.js
+++ b/dist/builder-manager/index.js
@@ -2993,12 +2993,8 @@ var Ue, he, si = /* @__PURE__ */ a(async (e) => {
     tsconfig: n,
     legalComments: "external",
     plugins: [
-      (0, vi.default)({
-        process: b.resolve("process/browser.js"),
-        util: b.resolve("util/util.js"),
-        assert: b.resolve("browser-assert")
-      }),
       Ve(qs),
+      require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')),
       (0, gi.pnpPlugin)()
     ],
     banner: {
@@ -3007,10 +3003,13 @@ var Ue, he, si = /* @__PURE__ */ a(async (e) => {
     footer: {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
+    inject: [b.resolve('node-stdlib-browser/helpers/esbuild/shim')],
     define: {
+      process: "process",
       "process.env": JSON.stringify(i),
       ...js(i),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/common/index.cjs b/dist/common/index.cjs
index 47f3153b9b42a50861ce6b69773bb2022daa66c8..8d23d5cba6a8e7cca0c096e26a5b59c6d68f98b8 100644
--- a/dist/common/index.cjs
+++ b/dist/common/index.cjs
@@ -134618,7 +134618,7 @@ function wq(e) {
     zwe = !0, a({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: false,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
diff --git a/dist/common/index.js b/dist/common/index.js
index 506359da4cb75e8eb816cfd8eb80d9687785fc4f..7ffc106d70963f49d004eea0140926f687fcb224 100644
--- a/dist/common/index.js
+++ b/dist/common/index.js
@@ -134579,7 +134579,7 @@ function ice(e) {
     z3e = !0, a({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: false,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
diff --git a/package.json b/package.json
index 7ffed272f7b5721cabad198354fd961203a65a2a..31d8b32764996173a7d8937255c441844e22aca5 100644
--- a/package.json
+++ b/package.json
@@ -280,15 +280,14 @@
     "@storybook/csf": "^0.1.11",
     "@types/express": "^4.17.21",
     "better-opn": "^3.0.2",
-    "browser-assert": "^1.2.1",
     "esbuild": "^0.18.0 || ^0.19.0 || ^0.20.0 || ^0.21.0 || ^0.22.0 || ^0.23.0",
     "esbuild-register": "^3.5.0",
     "express": "^4.19.2",
     "jsdoc-type-pratt-parser": "^4.0.0",
+    "node-stdlib-browser": "^1.2.1",
     "process": "^0.11.10",
     "recast": "^0.23.5",
     "semver": "^7.6.2",
-    "util": "^0.12.5",
     "ws": "^8.2.3"
   },
   "devDependencies": {
