diff --git a/cli.js b/cli.js
index 8ea2b15db9b225df3065eff4e09b35ad623a657a..cf878b1682e77519301522fb5629a75fe4e15206 100755
--- a/cli.js
+++ b/cli.js
@@ -5,6 +5,7 @@ var path = require('path')
 
 var argv = require('minimist')(process.argv.slice(2))
 var dotenv = require('dotenv')
+var dotenvFlow = require('dotenv-flow')
 var dotenvExpand = require('dotenv-expand').expand
 
 function printHelp () {
@@ -44,15 +45,23 @@ if (argv.e) {
     paths.push(...argv.e)
   }
 } else {
-  paths.push('.env')
+  paths = ['.env']
 }
 
 if (argv.c) {
-  paths = paths.reduce((accumulator, path) => accumulator.concat(
-    typeof argv.c === 'string'
-      ? [`${path}.${argv.c}.local`, `${path}.local`, `${path}.${argv.c}`, path]
-      : [`${path}.local`, path]
-  ), [])
+  paths = paths.reduce(function (accumulator, input) {
+    var filepaths = dotenvFlow.listFiles({ path: input.replace(/\/(\..+)+/, ''), debug: argv.debug })
+    var filepath = path.resolve(path.relative(process.cwd(), input))
+    return accumulator.concat((
+      typeof argv.c === 'string'
+        ? [`${filepath}.${argv.c}.defaults`, filepath, `${filepath}.${argv.c}`, `${filepath}.local`, `${filepath}.${argv.c}.local`]
+        : [`${filepath}.defaults`, filepath, `${filepath}.local`]
+    ).filter(function (filepath) {
+      return filepaths.includes(filepath)
+    }))
+  }, [])
+} else {
+  paths = dotenvFlow.listFiles({ debug: argv.debug })
 }
 
 function validateCmdVariable (param) {
@@ -71,22 +80,25 @@ if (argv.v) {
     variables.push(...argv.v.map(validateCmdVariable))
   }
 }
+
+var vaultVariables = process.env.DOTENV_KEY ? vaultVariables = dotenv.config({ processEnv: {}, debug: argv.debug }) : {}
 var parsedVariables = dotenv.parse(Buffer.from(variables.join('\n')))
+var parsedFile = dotenvFlow.load(paths, { debug: argv.debug, override })
+var finalVariables = { parsed: Object.assign({}, parsedFile.parsed, vaultVariables.parsed, parsedVariables) }
+if (argv.expand !== false) {
+  dotenvExpand(finalVariables)
+}
 
 if (argv.debug) {
-  console.log(paths)
-  console.log(parsedVariables)
+  if (process.env.CI) {
+    console.log('Only variables from source .env* are printed in CI mode...')
+    console.log(parsedFile.parsed)
+  } else {
+    console.log(finalVariables.parsed)
+  }
   process.exit()
 }
 
-paths.forEach(function (env) {
-  var parsedFile = dotenv.config({ path: path.resolve(env), override })
-  if (argv.expand !== false) {
-    dotenvExpand(parsedFile)
-  }
-})
-Object.assign(process.env, parsedVariables)
-
 if (argv.p) {
   var value = process.env[argv.p]
   console.log(value != null ? value : '')
