diff --git a/lib/config/config-loader.js b/lib/config/config-loader.js
index 39908f79fa5b6ff526e9bfc0054813c527d59cf4..2d5eeb12808fb47312d74be47656003b3cd9879d 100644
--- a/lib/config/config-loader.js
+++ b/lib/config/config-loader.js
@@ -242,19 +242,7 @@ async function loadConfigFile(filePath, hasUnstableNativeNodeJsTSConfigFlag) {
 
 	if (isTS) {
 		if (hasUnstableNativeNodeJsTSConfigFlag) {
-			if (isNativeTypeScriptSupportEnabled()) {
-				return await dynamicImportConfig(filePath, fileURL, mtime);
-			}
-
-			if (!("typescript" in process.features)) {
-				throw new Error(
-					"The unstable_native_nodejs_ts_config flag is not supported in older versions of Node.js.",
-				);
-			}
-
-			throw new Error(
-				"The unstable_native_nodejs_ts_config flag is enabled, but native TypeScript support is not enabled in the current Node.js process. You need to either enable native TypeScript support by passing --experimental-strip-types or remove the unstable_native_nodejs_ts_config flag.",
-			);
+			return await dynamicImportConfig(filePath, fileURL, mtime);
 		}
 
 		if (!isDeno && !isBun) {
