diff --git a/build/readConfigFileAndSetRootDir.js b/build/readConfigFileAndSetRootDir.js
index e989961ba71bbea43eb293ebdc7739f00f6e85e7..7f55a5fdb0619401f43aaa07050d5f77717e7aab 100644
--- a/build/readConfigFileAndSetRootDir.js
+++ b/build/readConfigFileAndSetRootDir.js
@@ -11,6 +11,13 @@ function path() {
   };
   return data;
 }
+function url() {
+  const data = _interopRequireWildcard(require('url'));
+  url = function () {
+    return data;
+  };
+  return data;
+}
 function fs() {
   const data = _interopRequireWildcard(require('graceful-fs'));
   fs = function () {
@@ -101,7 +108,8 @@ async function readConfigFileAndSetRootDir(configPath) {
   let configObject;
   try {
     if (isTS) {
-      configObject = await loadTSConfigFile(configPath);
+      const configModule = await import(url().pathToFileURL(configPath).href);
+      configObject = configModule.default;
     } else if (isJSON) {
       const fileContent = fs().readFileSync(configPath, 'utf8');
       configObject = (0, _parseJson().default)(
