diff --git a/dist/bin/next b/dist/bin/next
index 40956e6e02ba7b435b1fb751899042886d88210e..77a131faa2010eb196fd12672137109af8093742 100755
--- a/dist/bin/next
+++ b/dist/bin/next
@@ -21,10 +21,10 @@ if (process.env.NEXT_RSPACK) {
     // silent rspack's schema check
     process.env.RSPACK_CONFIG_VALIDATE = 'loose-silent';
 }
-if (!_semver.default.satisfies(process.versions.node, ">=20.9.0", {
+if (!_semver.default.satisfies(process.versions.node, ">=18.20.8", {
     includePrerelease: true
 })) {
-    console.error(`You are using Node.js ${process.versions.node}. For Next.js, Node.js version "${">=20.9.0"}" is required.`);
+    console.error(`You are using Node.js ${process.versions.node}. For Next.js, Node.js version "${">=18.20.8"}" is required.`);
     process.exit(1);
 }
 // Start performance profiling after Node.js version is checked
