diff --git a/bin/index.cjs b/bin/index.cjs
index d0395f4f3f9e415065d539f1dc7cf14f1eb2c824..ce13973e4eb9c3858483445373249c20f2a7261d 100755
--- a/bin/index.cjs
+++ b/bin/index.cjs
@@ -1,8 +1,8 @@
 #!/usr/bin/env node
 
 const majorNodeVersion = parseInt(process.versions.node, 10);
-if (majorNodeVersion < 20) {
-  console.error('To run Storybook you need to have Node.js 20 or higher');
+if (majorNodeVersion < 18) {
+  console.error('To run Storybook you need to have Node.js 18 or higher');
   process.exit(1);
 }
 
diff --git a/dist/builder-manager/index.cjs b/dist/builder-manager/index.cjs
index d25c8bd1565cb9746c9ca42be8a3766287dfa3df..9054e933e1c958dcc297b64284ea1b74f5a616a1 100644
--- a/dist/builder-manager/index.cjs
+++ b/dist/builder-manager/index.cjs
@@ -2008,17 +2008,20 @@ var Vi = /^\/($|\?)/, D, I, he = /* @__PURE__ */ o(async (e) => {
     jsxImportSource: "react",
     tsconfig: r,
     legalComments: "external",
-    plugins: [oe(Ht.globalsModuleInfoMap), (0, Jt.pnpPlugin)()],
+    plugins: [oe(Ht.globalsModuleInfoMap), require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')), (0, Jt.pnpPlugin)()],
     banner: {
       js: "try{"
     },
     footer: {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
+    inject: [require.resolve('node-stdlib-browser/helpers/esbuild/shim')],
     define: {
+      process: "process",
       "process.env": JSON.stringify(a),
       ...(0, Bt.stringifyProcessEnvs)(a),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/builder-manager/index.js b/dist/builder-manager/index.js
index abe09a07cb1ee3b3bac4ca0480823c720b358235..399eaa2e99d7cf11358adc492b8edb688af223d4 100644
--- a/dist/builder-manager/index.js
+++ b/dist/builder-manager/index.js
@@ -2023,17 +2023,20 @@ var pr = /^\/($|\?)/, L, C, _t = /* @__PURE__ */ o(async (e) => {
     jsxImportSource: "react",
     tsconfig: r,
     legalComments: "external",
-    plugins: [Z(cr), (0, Ct.pnpPlugin)()],
+    plugins: [Z(cr), require('node-stdlib-browser/helpers/esbuild/plugin')(require('node-stdlib-browser')), (0, Ct.pnpPlugin)()],
     banner: {
       js: "try{"
     },
     footer: {
       js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
     },
+    inject: [require.resolve('node-stdlib-browser/helpers/esbuild/shim')],
     define: {
+      process: "process",
       "process.env": JSON.stringify(a),
       ...lr(a),
       global: "window",
+      Buffer: "Buffer",
       module: "{}"
     }
   };
diff --git a/dist/common/index.cjs b/dist/common/index.cjs
index 85790ccf1d5cb67bb844e5953a9a0c21ca1e5bf1..43d44399309284f93d2f204438fff3c21fd06f3b 100644
--- a/dist/common/index.cjs
+++ b/dist/common/index.cjs
@@ -18745,7 +18745,7 @@ function sn(t) {
     SE = !0, i({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: !1,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
diff --git a/dist/common/index.js b/dist/common/index.js
index 2cbde37aa2bd4c2b1884fe606a77015606051220..70d260893228a9e6054d8b8c59adb0f57e6b8872 100644
--- a/dist/common/index.js
+++ b/dist/common/index.js
@@ -18676,7 +18676,7 @@ function dg(t) {
     Ql = !0, i({
       target: `node${process.version.slice(1)}`,
       format: "cjs",
-      hookIgnoreNodeModules: !0,
+      hookIgnoreNodeModules: !1,
       // Some frameworks, like Stylus, rely on the 'name' property of classes or functions
       // https://github.com/storybookjs/storybook/issues/19049
       keepNames: !0,
