diff --git a/dist/src/plugins/typescript/serviceCache.js b/dist/src/plugins/typescript/serviceCache.js
index f50d655abd2de3c394f8e3bbb5a85f7a3562342f..c8a9693055a2855579be06e0afb8b2abf197eaca 100644
--- a/dist/src/plugins/typescript/serviceCache.js
+++ b/dist/src/plugins/typescript/serviceCache.js
@@ -27,7 +27,7 @@ function createProject(host, createLanguageService, options) {
     if (major < 5) {
         return undefined;
     }
-    const factory = minor < 3 ? project_1.createProject : minor < 5 ? project_2.createProject : project_3.createProject;
+    const factory = major === 5 && minor < 3 ? project_1.createProject : major === 5 && minor < 5 ? project_2.createProject : project_3.createProject;
     const project = factory(typescript_1.default, host, createLanguageService, options);
     const proxyMethods = [
         'getCachedExportInfoMap',
