diff --git a/dist/index.js b/dist/index.js
index a2b3c64c80067db4b49372332f6ba1180ee25d27..9d1e7048ebfaea3224ba8d424dbde2d79254a30d 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -79,7 +79,7 @@ exports.DEFAULTS = {
 /**
  * TypeScript diagnostics error.
  */
-class TSError extends make_error_1.BaseError {
+class TSError extends Error {
     constructor(diagnosticText, diagnosticCodes, diagnostics = []) {
         super(`⨯ Unable to compile TypeScript:\n${diagnosticText}`);
         this.diagnosticCodes = diagnosticCodes;
@@ -95,12 +95,6 @@ class TSError extends make_error_1.BaseError {
             value: diagnostics,
         });
     }
-    /**
-     * @internal
-     */
-    [exports.INSPECT_CUSTOM]() {
-        return this.diagnosticText;
-    }
 }
 exports.TSError = TSError;
 const TS_NODE_SERVICE_BRAND = Symbol('TS_NODE_SERVICE_BRAND');
