diff --git a/dist/index.mjs b/dist/index.mjs
index 909ea96b2b4a5d7051690673d946536879e7613e..16b64545855921bd6cb4e5dea84f2be775aa7ad5 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -27,7 +27,8 @@ function sassGlobImports(options = {}) {
         for (let i2 = 0; i2 < searchBases.length; i2++) {
           basePath = searchBases[i2];
           files = globSync(path.join(basePath, globPattern), {
-            cwd: "./"
+            cwd: "./",
+            "windowsPathsNoEscape: true",
           }).sort((a, b) => a.localeCompare(b, "en"));
           const globPatternWithoutWildcard = globPattern.split("*")[0];
           if (globPatternWithoutWildcard.length) {
